src/main/resources/scripts/terminate_historic_completed_tasks_script.sql

```diff
@@ -0,0 +1,363 @@
+
+        SELECT t.state INTO task_state
+        FROM cft_task_db.tasks t
+        WHERE t.task_id = task_id_var;
+
+        -- Update the task state and other fields
+        UPDATE cft_task_db.tasks t
+        SET state = 'TERMINATED',
+            termination_reason = CASE
+                WHEN task_state = 'COMPLETED' THEN 'cancelled'
+                WHEN task_state = 'CANCELLED' THEN 'completed'
+            END,
+            last_updated_timestamp = NOW(),
+            last_updated_user = 'WA system user token in production',
+            last_updated_action = 'TerminateException'
+        WHERE task_id = task_id_var;
+
+        COMMIT;
+
+        -- Increment the counter for successfully updated tasks
+        updated_count := updated_count + 1;
+
+        -- Log the successfully updated task_id
+        RAISE INFO 'Successfully updated task_id: %, state: %, termination_reason: %, last_updated_timestamp: %, last_updated_user: %, last_updated_action: %',
+            task_id_var, 'TERMINATED',
+            task_state,
+            NOW(), 'WA system user token in production', 'TerminateException';
+    END LOOP;
+
+    -- Log the total count of successfully updated tasks
+    RAISE INFO 'Total successfully updated tasks: %', updated_count;
+END;
+$procedure$;
+
+-- The validate_terminated_tasks function validates the termination details of tasks in the cft_task_db.tasks table
+-- for a given array of task IDs. It checks if the termination_reason, last_updated_timestamp, last_updated_user,
+-- and last_updated_action fields meet specific criteria. For each task, it logs whether the conditions are valid or
+-- if any discrepancies are found. This function ensures that terminated tasks have consistent and accurate metadata.
+--
+-- Example for calling the function
+-- select cft_task_db.validate_terminated_tasks_in_primary(ARRAY['task_id1', 'task_id2', 'task_id3']);
+
+CREATE OR REPLACE FUNCTION cft_task_db.validate_terminated_tasks_in_primary(task_ids TEXT[])
+RETURNS VOID
+LANGUAGE plpgsql
+AS $function$
+DECLARE
+    task_id_var TEXT;
+    termination_reason_valid BOOLEAN;
+    timestamp_valid BOOLEAN;
+    user_valid BOOLEAN;
+    action_valid BOOLEAN;
+    state_valid BOOLEAN;
+BEGIN
+    -- Loop through each task_id in the array
+    FOR task_id_var IN SELECT UNNEST(task_ids)
+    LOOP
+        -- Initialize validation flags
+        state_valid := FALSE;
+        termination_reason_valid := FALSE;
+        timestamp_valid := FALSE;
+        user_valid := FALSE;
+        action_valid := FALSE;
+
+        -- Check the task details
+        SELECT
+            (state = 'TERMINATED') AS state_valid,
+            (termination_reason IN ('completed', 'cancelled')) AS termination_reason_valid,
+            (last_updated_timestamp >= NOW() - INTERVAL '5 minutes') AS timestamp_valid,
+            (last_updated_user = 'WA system user token in production') AS user_valid,
+            (last_updated_action = 'TerminateException') AS action_valid
```

Comment:
Again to make it more flexible we can provide the last_updated_user to look for and last_updated_timestamp interval as parameters.